pkg/ipcache/cidr.go

```diff
@@ -56,6 +57,7 @@ func AllocateCIDRs(
 		newlyAllocatedIdentities = map[string]*identity.Identity{}
 	}
 
+	IPIdentityCache.Lock()
```

Comment:
Yeah this part could trigger the bug fixed in https://github.com/cilium/cilium/pull/21565.